### DISCUSSION TITLE
Exporting STP file

### DISCUSSION THREAD

**Discussion by ericsuth4** [thliebig/openEMS-Project#230] [Q&A Usage]
using the python interface, can I export my design as an STP? I understand that I can write to an xml file, but I am unable to import this to other solvers.

I want to import this to CST or HFSS to verify some results.

**Comment by VolkerMuehlhaus** [comment]
You can export STL from AppCSXCAD, which is a widely supported 3D geometry format. STL export in this case will create multiple files, one for each material property.

If HFSS and CST don't support STL, you can also convert the files. Some 3D CAD programs like FreeCAD support both STL and STEP.

**Reply by ericsuth4** [reply]
Thanks for the response! the stl's work in HFSS. But is there a function to export them from the python interface? I want to automate the design and export process for some larger models.

I see the Write2xml function, but haven't been able to find one for stl.

**Reply by thliebig** [reply]
No I don't think that there is such an export at this point. But python can open AppCSXCAD with an extra argument to automatically export STL... Maybe that would help...
